client/src/component/modal/Modal.ts

```diff
@@ -58,19 +71,36 @@ class Modal extends core.RootComponent {
       const openForm = qs(openFormClass);
       openForm.classList.add('on');
     });
-  }
-  mounted() {
-    const categoryElem = qs('#card-category');
-    categoryElem.addEventListener('change', e => {
+    this.addEvent('submit', '.card-create', event => {
+      event.preventDefault();
+      const target = event.target as HTMLFormElement;
+      if (!target.category.value || !target.nickname.value.length) {
+        const alertWrapper = qs('#alert-wrapper');
+        new Alert(alertWrapper, 'alert-wrapper', {
+          text: constant.INPUT_REQUIRED,
+          color: 'danger',
+        });
+        return;
+      }
+      const payload = {
+        name: target.nickname.value,
+        cardCategoryId: target.category.value,
+      };
+      store.commit({ type: 'createCard', stateName: 'userCards', value: payload });
+    });
+    this.addEvent('change', '.category-wrapper', e => {
       /* @ts-ignore */
-      const value = parseInt(e.target.value);
-      let cardName = '';
-      this.$state.cardCategories.forEach((category: { id: number; name: string }) => {
-        if (category.id === value) cardName = category.name;
-      });
-      const cardBox = qs('.card-box');
-      cardBox.dataset.category = constant.CARD_ENG[cardName];
+      this.cardChange(e);
+    });
+  }
+  cardChange(e: { target: HTMLInputElement }) {
+    const value = parseInt(e.target.value);
+    let cardName = '';
+    this.$state.cardCategories.forEach((category: { id: number; name: string }) => {
+      if (category.id === value) cardName = category.name;
     });
+    const cardBox = qs('.card-box');
+    cardBox.dataset.category = constant.CARD_ENG[cardName];
   }
 }
```

Comment:
```suggestion
cardChange(e: Event) {
    const target = e.target as HTMLInputElement;
    const value = parseInt(target.value);
    let cardName = '';
    this.$state.cardCategories.forEach((category: { id: number; name: string }) => {
      if (category.id === value) cardName = category.name;
    });
    const cardBox = qs('.card-box');
    cardBox.dataset.category = constant.CARD_ENG[cardName];
  }
```

cardChange를 사용하는 부분의 ts-ignore를 없애고, 이런 식으로 as로 타입을 명시해줘서 사용하는 것도 좋을 것 같네요.. 어떤게 좋을까요 ?? as를 사용하는 것도 엄청 좋은 방법은 아닌 것 같지만 저는 lint를 무시하는 옵션보다는 나은 것 같아요!

Reply:
저도 동의합니다 ㅎㅎ 한번 수정 해보겠습니다